TESTING.md

```diff
@@ -0,0 +1,115 @@
+
+When contributing code to Prometheus-Operator, you'll notice that every Pull Request will run against an extensive test suite. Among an extensive list of benefits that tests brings to the Project's overall health and reliability, it can be the reviewer's and contributors's best friend during development:
+
+* Test cases serve as documentation, providing insights into the expected behavior of the software.
+* Testing can prevent regressions by verifying that new changes don't break existing functionality.
+* Running tests locally accelerate the feedback loop, removing the dependency that contributors might have on CI when working on a Pull Request.
+
+This document will focus on teaching you about the different test suites that we currently have and how to run different scenarios to help your development experience!
+
+# Test categories
+
+## Unit tests
+
+Unit tests are used to test particular code snippets in isolation. They are your best ally when looking for quick feedback loops in a particular function. 
+
+Imagine you're working on a PR that adds a new field to the Prometheus CRD and you want to test if your change is reflected to the statefulset. Instead of creating a full Kubernetes cluster, installing all the CRDs, running the Prometheus-Operator, deploying a Prometheus resource and finally check if your change made it to the live object, you could simply write or extend a unit test for the statefulset generation.
```

Comment:
(nit) I would have chosen the scenario where you add a new field to the Prometheus configuration because checking the generated config is enough in most cases it doesn't depend on the Kubernetes env. But feel free to ignore me :)

Reply:
Good point :) 

I've updated the example with tests for ScrapeConfig's string generation